source/Calamari/Commands/ReleasePackageLockCommand.cs

```diff
@@ -31,10 +26,27 @@ public ReleasePackageLockCommand(IVariables variables, IManagePackageUse package
             this.variables = variables;
             this.log = log;
             this.packageJournal = packageJournal;
+            timeBeforePackageExpiration = GetTimeBeforeExpiration(variables.Get(KnownVariables.Calamari.PackageRetentionLockExpiration));
             Options.Add("packageId=", "Package ID to download", v => packageId = v);
             Options.Add("packageVersion=", "Package version to download", v => packageVersion = v);
         }
 
+        TimeSpan GetTimeBeforeExpiration(string variable)
+        {
+            var defaultTime = TimeSpan.FromDays(14);
+            if (string.IsNullOrEmpty(variable)) return defaultTime;
+
+            try
+            {
+                return TimeSpan.Parse(variable);
```

Comment:
If you use `TimeSpan.TryParse(variable, out var timeBeforeExpiration)` (which returns a bool indicating success or failure) and an if-statement, you can get rid of the try-catch here.

Reply:
Nice! It's much cleaner.